tests/integration/events.go

```diff
@@ -148,52 +150,76 @@ func waitForUnsubscribeToCollectionEvent(s *state, action UnsubscribeToCollectio
 // update event to the local event bus.
 //
 // Expected document heads will be updated for any connected nodes.
-func waitForUpdateEvents(s *state, nodeID immutable.Option[int], collectionID int) {
+func waitForUpdateEvents(
+	s *state,
+	nodeID immutable.Option[int],
+	updates map[string]struct{},
+) {
 	for i := 0; i < len(s.nodes); i++ {
 		if nodeID.HasValue() && nodeID.Value() != i {
 			continue // node is not selected
 		}
 
-		var evt event.Update
-		select {
-		case msg, ok := <-s.nodeEvents[i].update.Message():
-			if !ok {
-				require.Fail(s.t, "subscription closed waiting for update event")
-			}
-			evt = msg.Data.(event.Update)
-
-		case <-time.After(eventTimeout):
-			require.Fail(s.t, "timeout waiting for update event")
+		expect := make(map[string]struct{})
```

Comment:
todo: Please set the length in the `make` call instead of resizing the map for each item.